packages/react-component-library/src/components/DatePicker/useOpenClose.ts

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from 'react'
+
+function hasParentWithMatchingSelector(target: Node, selector: string): boolean {
+  return [...document.querySelectorAll(selector)].some(
+    el => el !== target && el.contains(target)
+  )
+}
+
+export function useOpenClose(ref: React.RefObject<HTMLDivElement>) {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+
+  function handleClick(e: MouseEvent) {
```

Comment:
The thing is this stuff is covered by the following test:

https://github.com/Royal-Navy/standards-toolkit/pull/448/files#diff-1b225051935f3e919127b8f0d38191beR125

Do you think I need to test the hook itself independent of the component? Is that now a test of implementation detail?

Reply:
I am not necessarily a slave to the coverage metric but what I would have concerns about is that if I change the hook to what's below then all tests still pass.

I agree that we need to be careful of testing implementation details but I will favour getting the coverage right to prevent regressions.

```
import { useState, useEffect } from 'react'

export function useOpenClose() {
  const [isOpen, setIsOpen] = useState<boolean>(false)

  useEffect(() => {
    document.addEventListener('mousedown', () => {})

    return () => {
      document.removeEventListener('mousedown', () => {})
    }
  }, [])

  const onFocus = () => {
    setIsOpen(true)
  }

  return {
    isOpen,
    onFocus
  }
}
```

Reply:
Looks like the test I thought was covering that behaviour isn't. I'll dig a little deeper.